### ISSUE TITLE
Various issues bootstrapping from README.md

### ISSUE THREAD

**Issue by lamby** [WDDCP/commcare-wddcp#31]
(Migrating from PR #30)

A quick summary is that I filed PR #30 as this looked — on the face of it — like a Django compatibility problem, but after a quick chat/think we thought that something else is up. See also #28 which is suspicious.

The latest update is that:

 * Yesterday @DeckOfPandas set the Xenial ``box_version`` in https://github.com/DeckOfPandas/commcare-wddcp/commit/5e60bc6efbca4fc3d0daaacbb087941d57e16515
 * I blew away my vagrant yesterday with ``vagrant destroy`` to try @DeckOfPandas branch with the above commit on, but hit the fact that #27 was not merged on that branch. This resulted in an error during ``./scripts/docker runserver --bootstrap``
 * I cherry-picked that commit onto my local branch, ran ``destroy``, ``up`` etc. Then the ``./scripts/docker runserver --bootstrap`` failed after a 90m or so due to, hopefully, transient network/downloading problems
 * Now running again this morning. I have called ``destroy`` → ``up`` successfully and now executing ``./scripts/docker runserver --bootstrap`` again.

**Comment by lamby** [comment]
![](https://i.imgur.com/j2dj1ua.jpg)

Has hung here after 90m, trying again...

**Comment by lamby** [comment]
Error'd again. Full transcript is here: https://gist.github.com/lamby/8c801010971ff8d108e179f2b37ddb9e/raw

Tail of this log is:

```
  File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/mnt/commcare-hq/urls.py", line 167, in <module>
    urlconf_module, app_name, namespace = debug_toolbar_urls
ValueError: need more than 2 values to unpack
```

Trying again with https://github.com/WDDCP/commcare-wddcp/pull/30/commits/cdceba992a151f6faf2253efa0863cb3509718e0 ...

**Comment by lamby** [comment]
```
[...]
Step 2 : RUN plugin -i mobz/elasticsearch-head
 ---> Running in 98cc2f6f2468
-> Installing mobz/elasticsearch-head...
Trying https://github.com/mobz/elasticsearch-head/archive/master.zip...
Downloading ....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................DONE
Installed mobz/elasticsearch-head into /usr/share/elasticsearch/plugins/head
 ---> fe327c9002b5
Removing intermediate container 98cc2f6f2468
Step 3 : COPY files/elasticsearch.yml /usr/share/elasticsearch/config/elasticsearch.yml
 ---> 4363d3fe0645
Removing intermediate container 7f443b9195aa
Successfully built 4363d3fe0645
WARNING: Image for service elasticsearch was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Pulling kafka (spotify/kafka:latest)...
latest: Pulling from spotify/kafka
386a066cd84a: Pull complete
75ea84187083: Pull complete
3e2e387eb26a: Pull complete
eef540699244: Pull complete
1624a2f8d114: Downloading
7018f4ec6e0a: Downloading [===========================>                       ]    29 MB/53.45 MB
6ca3bc2ad3b3: Download complete
658810c91c6a: Downloading [=======================================>           ] 50.54 MB/63.18 MB
c8bd0e6d651a: Downloading
d471f5c9f3f2: Waiting
ERROR: dial tcp: lookup auth.docker.io on 10.0.2.3:53: read udp 10.0.2.15:52229->10.0.2.3:53: i/o timeout
```

Running again...

**Comment by lamby** [comment]
Identical failure

```
ERROR: dial tcp: lookup auth.docker.io on 10.0.2.3:53: read udp 10.0.2.15:52229->10.0.2.3:53: i/o timeout
```

Trying again...

**Comment by lamby** [comment]
Hung here:

```
Digest: sha256:8f6ed2c21e255a7e49ffd57272d2b0db5067005d79839972d89c109e266a8633
Status: Downloaded newer image for dimagi/docker-postgresql:latest
Pulling redis (redis:latest)...
latest: Pulling from library/redis
065132d9f705: Pull complete
fc32c7d9b0f4: Pull complete
ad60cc6fa431: Pull complete
b21c99d8cf03: Pull complete
357908014789: Pull complete
e27e1cb0ca43: Pull complete
Digest: sha256:fe77356e6e8d5c5200b9800e50ae71147efdc446a3cc4f601c607fbfd218015e
Status: Downloaded newer image for redis:latest
Pulling couch (klaemo/couchdb:2.0-dev)...
2.0-dev: Pulling from klaemo/couchdb
aa18ad1a0d33: Downloading [====>                                              ] 4.325 MB/52.6 MB
b75645159bde: Download complete
3baed85ac4a5: Downloading [=>                                                 ] 2.685 MB/92.3 MB
7498ab9c7fce: Download complete
d890d2b4b7ab: Waiting
83835fba49e0: Waiting
```

**Comment by lamby** [comment]
Another hang:

```
ubuntu@ubuntu-xenial:~$ cd /vagrant
ubuntu@ubuntu-xenial:/vagrant$ ./scripts/docker runserver --bootstrap
Pulling postgres (dimagi/docker-postgresql:latest)...
latest: Pulling from dimagi/docker-postgresql
51f5c6a04d83: Pull complete
a3ed95caeb02: Pull complete
4a40f149a638: Pull complete
7e8f4bc7ac43: Downloading [======================================>            ] 938.6 kB/1.216 MB
f2a70e1bea0a: Download complete
6d6ccf3c012c: Download complete
4b0c5c71fa5d: Download complete
ff64a4df1891: Download complete
d1f8420bd4aa: Download complete
5016d23c63ab: Download complete
d5db09d384ec: Download complete
055e85b433f4: Download complete
8cf0092885bb: Download complete
49444b73b39a: Download complete
c160824a4568: Download complete
```

**Comment by lamby** [comment]
And again:

```
21c99d8cf03: Pull complete
357908014789: Pull complete
e27e1cb0ca43: Pull complete
Digest: sha256:fe77356e6e8d5c5200b9800e50ae71147efdc446a3cc4f601c607fbfd218015e
Status: Downloaded newer image for redis:latest
Pulling couch (klaemo/couchdb:2.0-dev)...
2.0-dev: Pulling from klaemo/couchdb
aa18ad1a0d33: Pull complete
b75645159bde: Pull complete
3baed85ac4a5: Pull complete
7498ab9c7fce: Pull complete
d890d2b4b7ab: Pull complete
83835fba49e0: Pull complete
Digest: sha256:5fc129e16ea16a0d65a7b5f762e6fee29bf43576a13d24ae1d4d72889e6b581d
Status: Downloaded newer image for klaemo/couchdb:2.0-dev
Building elasticsearch
Step 1 : FROM elasticsearch:1.7.4
1.7.4: Pulling from library/elasticsearch
03e1855d4f31: Downloading [=>                                                 ] 1.042 MB/51.36 MB
a3ed95caeb02: Download complete
9269ba3950bb: Download complete
785ac756bcc8: Download complete
6fed5a9244a3: Download complete
cafd70c0e522: Download complete
58d01b4a63cc: Downloading [================================>                  ] 34.46 MB/53.33 MB
b9e1d590fa13: Download complete
a640017432b9: Downloading
9b65bdd93f26: Waiting
127973a59117: Waiting
169c8e1d35ad: Waiting
1c7bd17b58cb: Waiting
f9379a3ee102: Waiting
2e394ed375cf: Waiting
d68b48d9197c: Waiting
ERROR: Service 'elasticsearch' failed to build: dial tcp: lookup auth.docker.io on 10.0.2.3:53: read udp 10.0.2.15:43148->10.0.2.3:53: i/o timeout
```

**Comment by lamby** [comment]
After another failure identical to the one above, I have successfully bootstrapped it:

https://gist.github.com/lamby/47704c925aa2e78c7014b3e6b9320734/raw

**Comment by lamby** [comment]
Note in particular that it is using Django 1.10.7, this is specified in:

https://github.com/WDDCP/commcare-wddcp/blob/wddcp/requirements/requirements.txt#L10

… which actually suggests my PR regarding updating our ``urls.py`` to match (#30) is still valid, so I'm going to reopen that now. Not sure what to do with this issue atm.

**Comment by lamby** [comment]
Okay, I'm seeing my "original" issue:

```
ubuntu@ubuntu-xenial:/vagrant$ ./scripts/docker bash
running: bash
root@ef2c35361476:/mnt/commcare-hq# cd /mnt/commcare-hq
root@ef2c35361476:/mnt/commcare-hq# ./manage.py
Traceback (most recent call last):
  File "./manage.py", line 118, in <module>
    execute_from_command_line(sys.argv)
  File "/vendor/lib/python2.7/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/vendor/lib/python2.7/site-packages/django/core/management/__init__.py", line 341, in execute
    django.setup()
  File "/vendor/lib/python2.7/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/vendor/lib/python2.7/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/vendor/lib/python2.7/site-packages/django/apps/config.py", line 90, in create
    module = import_module(entry)
  File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named djtables
```

Is it installed anywhere in any virtualenv?

```
root@ef2c35361476:/mnt/commcare-hq# find / | grep -i djtables | wc -l
0
```
(Nope. )

**Comment by lamby** [comment]
Ah, progress: ``./scripts/docker runserver `` must be run concurrently (and actually complete) when you attempt the ``makemigrations`` (etc.) step. This needs documenting -- specifically, the README.md needs re-ordering to not suggest that the "Run" section is later.

**Comment by lamby** [comment]
![](https://i.imgur.com/c7Xh5Df.jpg)

**Comment by DeckOfPandas** [comment]
(Probably) fixed in 894603b168b23eb6b0d0fc34542bf8d9f7bc8ef0
